core/src/main/clojure/xtdb/indexer.clj

```diff
@@ -571,14 +580,17 @@
                               !xtql-idxer (delay (->xtql-indexer allocator live-idx-tx tx-ops-rdr ra-src wm-src scan-emitter tx-opts))
                               !sql-idxer (delay (->sql-indexer allocator live-idx-tx tx-ops-rdr ra-src wm-src scan-emitter tx-opts))]
                           (dotimes [tx-op-idx (.valueCount tx-ops-rdr)]
-                            (when-let [more-tx-ops (case (.getLeg tx-ops-rdr tx-op-idx)
-                                                     :xtql (.indexOp ^OpIndexer @!xtql-idxer tx-op-idx)
-                                                     :sql (.indexOp ^OpIndexer @!sql-idxer tx-op-idx)
-                                                     :put-docs (.indexOp ^OpIndexer @!put-docs-idxer tx-op-idx)
-                                                     :delete-docs (.indexOp ^OpIndexer @!delete-docs-idxer tx-op-idx)
-                                                     :erase-docs (.indexOp ^OpIndexer @!erase-docs-idxer tx-op-idx)
-                                                     :call (.indexOp ^OpIndexer @!call-idxer tx-op-idx)
-                                                     :abort (throw abort-exn))]
+                            (when-let [more-tx-ops
+                                       (.recordCallable tx-timer
+                                                        (fn []
+                                                          (case (.getLeg tx-ops-rdr tx-op-idx)
+                                                            :xtql (.recordCallable xtql-timer #(.indexOp ^OpIndexer @!xtql-idxer tx-op-idx))
+                                                            :sql (.recordCallable sql-timer #(.indexOp ^OpIndexer @!sql-idxer tx-op-idx))
+                                                            :put-docs (.recordCallable put-docs-timer #(.indexOp ^OpIndexer @!put-docs-idxer tx-op-idx))
+                                                            :delete-docs (.recordCallable delete-docs-timer #(.indexOp ^OpIndexer @!delete-docs-idxer tx-op-idx))
+                                                            :erase-docs (.recordCallable erase-docs-timer #(.indexOp ^OpIndexer @!erase-docs-idxer tx-op-idx))
+                                                            :call (.recordCallable call-timer #(.indexOp ^OpIndexer @!call-idxer tx-op-idx))
```

Comment:
@jarohen As `call` might be recursive, this might double counting things.

Reply:
yeah :+1: 100% accuracy probably isn't required here, so not too worried about that tbh. 

To your point about granularity, now that you have the wiring set up it might be a good idea to talk about what metrics you think we should have?